opacus/privacy_engine.py

```diff
@@ -29,29 +42,43 @@ def forbid_accumulation_hook(module: nn.Module, _):
 
 
 class PrivacyEngine:
+    """
+    Main entry point to the Opacus API - use ``PrivacyEngine``  to enable differential
+    privacy for your model training.
+
+    ``PrivacyEngine`` object encapsulates current privacy state (privacy budget +
+    method it's been calculated) and exposes ``make_private`` method to wrap your
+    PyTorch training objects with their private counterparts.
+
+    Example:
+        >>> dataloader = demo_dataloader
+        >>> model = MyCustomModel()
+        >>> optimizer = torch.optim.SGD(model.parameters(), lr=0.05)
+        >>> privacy_engine = PrivacyEngine()
+        >>>
+        >>> model, optimizer, dataloader = privacy_engine.make_private(
+        ...    module=model,
+        ...    optimizer=optimizer,
+        ...    data_loader=dataloader,
+        ...    noise_multiplier=1.0,
+        ...    max_grad_norm=1.0,
+        ... )
+        >>> # continue training as normal
```

Comment:
Does this work with doctest? @romovpa

Reply:
Yes, you can see this is executed and passed in the unit test run: https://app.circleci.com/pipelines/github/pytorch/opacus/1324/workflows/be756a10-42ec-43e0-ac32-999d13bb06fa/jobs/6942

```
opacus/per_sample_gradient_clip.py s                                     [  0%]
opacus/privacy_engine.py .                                               [  1%]
opacus/accountants/analysis/rdp.py .                                     [  2%]
opacus/grad_sample/utils.py .                                            [  3%]
opacus/layers/dp_multihead_attention.py .                                [  3%]
opacus/layers/param_rename.py .                                          [  4%]
opacus/tests/accountants_test.py ...                                     [  7%]
<...>
```